wpilibjIntegrationTests/src/main/java/edu/wpi/first/wpilibj/RobotDriveTest.java

```diff
@@ -0,0 +1,175 @@
+  private FakeSpeedController m_rdFrontRight = new FakeSpeedController();
+  private FakeSpeedController m_rdRearRight = new FakeSpeedController();
+  private FakeSpeedController m_frontLeft = new FakeSpeedController();
+  private FakeSpeedController m_rearLeft = new FakeSpeedController();
+  private FakeSpeedController m_frontRight = new FakeSpeedController();
+  private FakeSpeedController m_rearRight = new FakeSpeedController();
+  private RobotDrive m_robotDrive =
+      new RobotDrive(m_rdFrontLeft, m_rdRearLeft, m_rdFrontRight, m_rdRearRight);
+  private DifferentialDrive m_differentialDrive =
+      new DifferentialDrive(m_frontLeft, m_frontRight);
+  private MecanumDrive m_mecanumDrive =
+      new MecanumDrive(m_frontLeft, m_rearLeft, m_frontRight, m_rearRight);
+
+  private final double[] m_testJoystickValues = {1.0, 0.5, 0.0, -0.5, -1.0};
+  private final double[] m_testGyroValues = {0, 45, 90, 135, 180, 225, 270, 305, 360, 540,
+                                             -45, -90, -135, -180, -225, -270, -305, -360, -540};
+
+  @Override
+  protected Logger getClassLogger() {
+    return logger;
+  }
+
+  @Test
+  public void testTankDriveSquared() {
+    for (double leftJoystick : m_testJoystickValues) {
+      for (double rightJoystick : m_testJoystickValues) {
+        m_robotDrive.tankDrive(leftJoystick, rightJoystick);
+        m_differentialDrive.tankDrive(leftJoystick, rightJoystick);
+        assertEquals("Left Motor squared didn't match. Left Joystick: " + leftJoystick
+            + " Right Joystick: " + rightJoystick + " ", m_rdFrontLeft.get(), m_frontLeft.get(),
+            0.01);
+        assertEquals("Right Motor squared didn't match. Left Joystick: " + leftJoystick
+            + " Right Joystick: " + rightJoystick + " ", m_rdFrontRight.get(), m_frontRight.get(),
+            0.01);
+      }
+    }
+  }
+
+  @Test
+  public void testTankDrive() {
+    for (double leftJoystick : m_testJoystickValues) {
+      for (double rightJoystick : m_testJoystickValues) {
+        m_robotDrive.tankDrive(leftJoystick, rightJoystick, false);
+        m_differentialDrive.tankDrive(leftJoystick, rightJoystick, false);
+        assertEquals("Left Motor didn't match. Left Joystick: " + leftJoystick
+            + " Right Joystick: " + rightJoystick + " ", m_rdFrontLeft.get(), m_frontLeft.get(),
+            0.01);
+        assertEquals("Right Motor didn't match. Left Joystick: " + leftJoystick
+            + " Right Joystick: " + rightJoystick + " ", m_rdFrontRight.get(), m_frontRight.get(),
+            0.01);
+      }
+    }
+  }
+
+
+  @Test
+  public void testArcadeDriveSquared() {
+    for (double moveJoystick : m_testJoystickValues) {
+      for (double rotateJoystick : m_testJoystickValues) {
+        m_robotDrive.arcadeDrive(moveJoystick, rotateJoystick);
+        m_differentialDrive.tankDrive(moveJoystick, rotateJoystick);
+        assertEquals("Left Motor squared didn't match. Move Joystick: " + moveJoystick
```

Comment:
What are all these "Left Motor...didn't match?"

Reply:
They are messages that will be printed if the test fails. See http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(java.lang.String,%20double,%20double,%20double)

Reply:
Ah, didn't realize it was only if it fails. Does the assertionerror not print out expected vs actual on its own?

Reply:
It prints actual vs expected, but this provides the inputs to the functions.